### PR TITLE
go: update to urfave v2, merge endpoint-monitor and op-heartbeat into main go module

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

- Merge endpoint-monitor into root Go module
  - Updated to no longer depend on legacy l2geth.
- Merge op-heartbeat into root Go module
- Update all main Go-mod packages to urfave CLI v2.
   - op-batcher
   - op-bootnode
   - op-challenger
   - op-heartbeat
   - op-node
   - op-program/host
   - op-proposer
   - op-service
   - op-signer/client
   - op-wheel

Cannon + Geth are already using urfave CLI v2.

With this upgrade, in separate later PRs, we can improve op-service without complicating a later v2 migration, and utilize the improved context/shutdown functionality of urfave CLI v2.

In practice, most of the v2 CLI migration changes are:
- commands/flags now with pointers
- EnvVars instead of EnvVar
- import-path updates

**Tests**

No functionality changes

**Metadata**

Fix CLI-3946

### CODE REVIEW

File: op-service/log/cli.go [hunk 1]
```diff
@@ -93,10 +93,10 @@ func ReadLocalCLIConfig(ctx *cli.Context) CLIConfig {
 
 func ReadCLIConfig(ctx *cli.Context) CLIConfig {
```
Comment (trianglesphere) [inline op-service/log/cli.go:94]: Can we deprecate `ReadLocalCLIConfig`? Do we have any other spots where we had a difference between local & global?

**Review by trianglesphere — CHANGES_REQUESTED** [review]
I think the flipped CheckRequired is the reason the tests are failing.
Lots of code to review, but it was pretty straightforwards.

File: op-batcher/compressor/cli.go [hunk 2]
```diff
@@ -15,30 +15,33 @@ const (
 )
 
 func CLIFlags(envPrefix string) []cli.Flag {
+	prefixEnvVars := func(name string) []string {
```
Comment (tynes) [inline op-batcher/compressor/cli.go:18]: Could this function be in `op-service`?

File: op-batcher/flags/flags.go [hunk 3]
```diff
@@ -110,8 +114,8 @@ var Flags []cli.Flag
 
 func CheckRequired(ctx *cli.Context) error {
 	for _, f := range requiredFlags {
-		if !ctx.GlobalIsSet(f.GetName()) {
-			return fmt.Errorf("flag %s is required", f.GetName())
+		if !ctx.IsSet(f.Names()[0]) {
```
Comment (tynes) [inline op-batcher/flags/flags.go:117]: Could this out of bounds?

**Review by tynes — COMMENTED** [review]
Generally looks good to me

File: op-batcher/compressor/cli.go (hunk 2 above)
Comment (protolambda) [reply op-batcher/compressor/cli.go:18]: there already is `opservice.PrefixEnvVar`. We can make that output a string-slice, but we'd still need to bind it to `envPrefix`, so it doesn't get any shorter, or we have to repeat `envPrefix` for every call.

File: op-batcher/flags/flags.go (hunk 3 above)
Comment (protolambda) [reply op-batcher/flags/flags.go:117]: There's always at least 1 name in a flag

File: op-service/log/cli.go (hunk 1 above)
Comment (protolambda) [reply op-service/log/cli.go:94]: will remove `ReadLocalCLIConfig`, there are not many usages of it, and it shouldn't make a difference anymore.

File: op-node/flags/flags.go [hunk 4]
```diff
@@ -258,8 +258,8 @@ func init() {
 
 func CheckRequired(ctx *cli.Context) error {
 	for _, f := range requiredFlags {
-		if !ctx.GlobalIsSet(f.GetName()) {
-			return fmt.Errorf("flag %s is required", f.GetName())
+		if ctx.IsSet(f.Names()[0]) {
```
[the remark this answers is not in the document]
Comment (protolambda) [reply op-node/flags/flags.go:261]: ty, nice catch

File: op-batcher/compressor/cli.go (hunk 2 above)
Comment (trianglesphere) [reply op-batcher/compressor/cli.go:18]: we didn't have a problem repeating `envPrefix`. IMO `opservice.PrefixEnvVar` should return a string slice now

Comment (protolambda) [reply op-batcher/compressor/cli.go:18]: And rename it to?

Comment (protolambda) [reply op-batcher/compressor/cli.go:18]: Also I don't like repeating envPrefix, seems more error prone :/

Comment (protolambda) [reply op-batcher/compressor/cli.go]: updated to return in a slice

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**